unity-client/Assets/Scripts/MainScripts/DCL/Components/BuilderInWorld/DCLName.cs

```diff
@@ -0,0 +1,38 @@
+using DCL;
+using DCL.Components;
+using DCL.Controllers;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// This component is a descriptive name of the Entity. In the BuilderInWorld you can give an entity a descriptive name through the entity list.
+/// Builder in World send a message to kernel to change the value of this component in order to assign a descriptive name
+/// </summary>
+public class DCLName : BaseDisposable
+{
+    [System.Serializable]
+    public class Model
+    {
+        public string value;
+    }
+
+    public Model model;
+
+    public DCLName(ParcelScene scene) : base(scene)
+    {
+        model = new Model();
+    }
+
+    public override IEnumerator ApplyChanges(string newJson)
```

Comment:
I would add something like:
```
#if UNITY_EDITOR
foreach (DecentralandEntity decentralandEntity in this.attachedEntities)
                decentralandEntity.meshRootGameObject.name += model.value;
#endif
```
If we are gonna have named entities, we might as well be able to find them in the hierarchy =).

Be aware that the code above is just a sample, probably you will have to ensure that multiple `ApplyChanges` are processed correctly and the format could be way better.